### PR TITLE
 ci: add spell-checking with typos

### PR DESCRIPTION
**Pull request by not-my-profile**
This is the followup PR for #3458 to prevent accidental typos from ending up in the codebase.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -139,3 +139,12 @@ jobs:
       - name: Rebuild Wasm and verify it hasn't changed
         if: success() && steps.source.outputs.modified == 'true'
         run: deno task --cwd ${{ matrix.module }} wasmbuild --check
+
+  typos:
+    name: "spell check"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: crate-ci/typos@master
```
Comment (kt3k) [inline .github/workflows/ci.yml:148]: I think we should run this step as part of `lint` job

**Comment by kt3k** [comment]
The step seems taking only 1 sec (ref. https://github.com/denoland/deno_std/actions/runs/5336711497/jobs/9671774578 ). I'm in favor of including this

File: .github/workflows/ci.yml (hunk 1 above)
Comment (not-my-profile) [reply .github/workflows/ci.yml]: sure

**Review by kt3k — APPROVED** [review]
LGTM

**Comment by kt3k** [comment]
Can you also add `typos` task in `deno.json` and update 'Contrbuting' section in README.md?

**Review by kt3k — APPROVED** [review]
LGTM again. Thanks!

**Pull request merged**